src/components/List/List.js

```diff
@@ -0,0 +1,162 @@
+        let colorsArray = [];
+        //set the colors array such that, Past date:red - Current date:blue - Future date:Purple  
+        for (let i = 0; i < filteredDataArray.length; i++) {
+            if (compareDates(filteredDataArray[i].date) === 1)
+                colorsArray[i] = "#982176"; //purple
+            else if (compareDates(filteredDataArray[i].date) === 0)
+                colorsArray[i] = "#1434A4"; //blue
+            else if (compareDates(filteredDataArray[i].date) === -1)
+                colorsArray[i] = "#D22B2B"; //red
+        }
+        setCardsColorsArray(colorsArray);
+    }
+
+    //This function is used when clicking a div to edit it's event
+    const divClickHandler = (event) => {
+        let divData = event.currentTarget.textContent;
+        var myArray = divData.trim().split("  ");
+        var name = myArray[0].toUpperCase();
+
+        for (let k = 0; k < filteredDataArray.length; k++) {
+            if (name === filteredDataArray[k].name.toUpperCase()) {
+                console.log(filteredDataArray[k].name);
+                set(ref(db, 'eventToEdit/'),
+                    {
+                        id: filteredDataArray[k].id,
+                        name: filteredDataArray[k].name,
+                        description: filteredDataArray[k].description,
+                        date: filteredDataArray[k].date
+                    });
+                break;
+            }
+        }
+        navigate('/form', { state: { id: 2, headerText: 'Edit Event', buttonText: 'Save' } }); //navigate To Form Page
+    }
+
+    //This function converts firebase snapshot to array
+    const snapshotToArray = (snapshot) => {
+        var returnArr = [];
+        snapshot.forEach(function (childSnapshot) {
+            var item = childSnapshot.val();
+            item.key = childSnapshot.key;
+            returnArr.push(item);
+        });
+        return returnArr;
+    }
+
+    //This function compares one date with the current date and returns 1 if it's a future date, 0 the current date, -1 a past date
+    const compareDates = (divDate2) => {
+        let currDate = new Date(); //current date format : 8/19/2023
+        let divDate = new Date(divDate2); //convert div date format from 2023-08-31T14:40   -  type="datetime-local" to date object format
+
+        let currDay = currDate.getDate();
+        let currMonth = currDate.getMonth() + 1;
+        let currYear = currDate.getFullYear();
+
+        let divDay = divDate.getDate();
+        let divMonth = divDate.getMonth() + 1;
+        let divYear = divDate.getFullYear();
+
+        if (divYear > currYear || (divYear === currYear && divMonth > currMonth) || (divYear === currYear && divMonth === currMonth && divDay > currDay)) {
+            return 1;
+        }
+        else if (divYear === currYear && divMonth === currMonth && divDay === currDay) {
+            return 0;
+        }
+        else {
+            return -1;
+        }
+    }
```

Comment:
compareDates, snapshotToArray are common functions, so please under services folder create folder called utility and under it create file called utils.js and then move these 2 function to it

Reply:
done